### ISSUE TITLE
Errors on initial atclient sync

### ISSUE THREAD

**Issue by cconstab** [atsign-foundation/at_client_sdk#842]
### Describe the bug

Exceptions thrown on initial sync 

### Steps to reproduce

1. Fist use the at_talk binary
2. Then delete the ~/.ai6bh (namespace) 
3. And then chat betwen the two atSigns
4. I have seen this error or similar also with sshnp using the -v

### Expected behavior

No exceptions or good reasons for them

### Screenshots

```
SEVERE|2022-12-08 15:43:30.943175|SyncService|exception syncing entry to local {atKey: public:cached:public:publickey@iot_manager, value: MIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAmVLEk3IbJlFxtX4gAfw3RmEuEAkG7fY3bARun9SPDE2j0m8SZFLi/Rnu/6YC5o7zsubpgec/aIgaLC+NTLvHWU2RmZeluinRUb0wT+mdv6uX67IXRPpbHSZsZDlgDyySEtn2NgFzBwxtdumtOHaJ5z2XpZdz4yeznULWcQlu/a8542liUan8kElIsmKTIYax2VCfcA+jSEfWyAxUE//hQT7svbvfV2Zv+igldaCVhO65viRjdGIX9+Nr6MAz66JtKpdy+jWS/tknMBEXh79dTMyC1G+52i1/TQXgPe64x2jplzFN8BEpZYY9BCoDZceqnJySZFSBvHp01/06MBY1SwIDAQAB, metadata: {createdAt: 2022-08-23 19:54:06.784Z, updatedAt: 2022-08-23 19:54:06.784Z, availableAt: 2022-08-23 19:54:06.784Z, status: active, version: 0, ttl: 0, ttb: 0, ccd: false, isBinary: false, isEncrypted: false}, commitId: 44458, operation: *} Exception: Exception: Key public:cachednull is invalid - stacktrace: #0      HiveKeystore.putAll (package:at_persistence_secondary_server/src/keystore/hive_keystore.dart:354)
#1      LocalSecondary._update (package:at_client/src/client/local_secondary.dart:101)
#2      LocalSecondary.executeVerb (package:at_client/src/client/local_secondary.dart:43)
#3      SyncServiceImpl._pullToLocal (package:at_client/src/service/sync_service_impl.dart:838)
#4      SyncServiceImpl._syncLocal (package:at_client/src/service/sync_service_impl.dart:767)
#5      SyncServiceImpl._syncFromServer (package:at_client/src/service/sync_service_impl.dart:462)
<asynchronous suspension>
#6      SyncServiceImpl.syncInternal (package:at_client/src/service/sync_service_impl.dart:341)
<asynchronous suspension>
#7      SyncServiceImpl.processSyncRequests (package:at_client/src/service/sync_service_impl.dart:215)
<asynchronous suspension>
#8      SyncServiceImpl._scheduleSyncRun.<anonymous closure> (package:at_client/src/service/sync_service_impl.dart:97)
<asynchronous suspension>
#9      _httpConnectionHook.<anonymous closure> (dart:_http/embedder_config.dart:21)
<asynchronous suspension>


WARNING|2022-12-08 15:43:30.944177|HiveKeystore|Key public:cachednull is invalid

SEVERE|2022-12-08 15:43:30.944177|SyncService|exception syncing entry to local {atKey: public:cached:public:publickey@iot_device01, value: MIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAs/d81xTUf/pZcmP7WD8pVYcPw3PrKzpC9ywPfLyDd7/5Rc5ylxlKWALTasH8QfEPSkI/3b2sKStQL1dKU94+fKLGeWKVHtf8P3UzUs+ERTF076w7W4GGjIMr/mjkcmpOycNEz60HAjlL8KMApEHTfNaQLYzgDJeCZMpVvEz/snIGkdeSmoLduT28jDiQ7JQDeV7RXPbxejePD2k2STnGgvVof7UPu28kUVRA6U1dWdbU8CyQ2pzZig/Hps5XD9nsWgUhMTkNIz5SKBqV4KN9ALeXiYgePRRR+exvJCqCYZW8/DxuPOcpK6iKBL4aWJj45zC2ZLgGDfhdQedGueuciQIDAQAB, metadata: {createdAt: 2022-08-23 19:54:07.203Z, updatedAt: 2022-08-23 19:54:07.203Z, availableAt: 2022-08-23 19:54:07.203Z, status: active, version: 0, ttl: 0, ttb: 0, ccd: false, isBinary: false, isEncrypted: false}, commitId: 44462, operation: *} Exception: Exception: Key public:cachednull is invalid - stacktrace: #0      HiveKeystore.putAll (package:at_persistence_secondary_server/src/keystore/hive_keystore.dart:354)
#1      LocalSecondary._update (package:at_client/src/client/local_secondary.dart:101)
#2      LocalSecondary.executeVerb (package:at_client/src/client/local_secondary.dart:43)
#3      SyncServiceImpl._pullToLocal (package:at_client/src/service/sync_service_impl.dart:838)
#4      SyncServiceImpl._syncLocal (package:at_client/src/service/sync_service_impl.dart:767)
#5      SyncServiceImpl._syncFromServer (package:at_client/src/service/sync_service_impl.dart:462)
<asynchronous suspension>
#6      SyncServiceImpl.syncInternal (package:at_client/src/service/sync_service_impl.dart:341)
<asynchronous suspension>
#7      SyncServiceImpl.processSyncRequests (package:at_client/src/service/sync_service_impl.dart:215)
<asynchronous suspension>
#8      SyncServiceImpl._scheduleSyncRun.<anonymous closure> (package:at_client/src/service/sync_service_impl.dart:97)
<asynchronous suspension>
#9      _httpConnectionHook.<anonymous closure> (dart:_http/embedder_config.dart:21)
<asynchronous suspension>
````

### Smartphones

- Dart on PC/Linux

### Were you using an atApplication when the bug was found?

at_talk and sshnp

**Comment by sitaram-kalluri** [comment]
@cconstab: Can you please let us know that atSign on which you have observed the issue?

Ran at_talk locally and was unable to reproduce the issue. I was able to chat between the atSign's successfully and removed `ai6bh` namespace and did not observe the issue on initial sync

The sync exception you mentioned was because of the malformed key "public:cached:public:publickey@iot_manager" which is reported (and fixed) - https://github.com/atsign-foundation/at_server/issues/865. 

We did add a validation layer in the persistent store to prevent malformed keys from getting into the secondary server. Also, we did have a fix that will scan for the existing malformed keys on server start-up and removes any key that starts with "public:cached". 

The fix is merged to the trunk and deployed into canary - 3.0.26b

https://user-images.githubusercontent.com/16645698/206682303-05229f18-293f-4c14-a602-78cf81b4444d.mp4

**Comment by cconstab** [comment]
Very cool! My atSigbs are all on prod so no doubt this will fix things

**Comment by sitaram-kalluri** [comment]
The changes are released into production in v3.0.26

**Comment by gkc** [comment]
Closing as fixed.
